### PR TITLE
Provide `renderWhenVisible` to LadTableSets

### LINKED ISSUE #7383: LAD Table Sets failing to initialize telemetry
#### Summary

LAD Table Sets are failing to initialize telemetry on load:
```js
Uncaught (in promise) TypeError: this.renderWhenVisible is not a function
    at Proxy.updateView (LadRow.vue:244:34)
    at Proxy.setLatestValues (LadRow.vue:267:12)
    at TelemetryCollection.emit (index.js:116:1)
    at TelemetryCollection._processNewTelemetry (TelemetryCollection.js:270:1)
    at TelemetryCollection._requestHistoricalTelemetry (TelemetryCollection.js:172:1)
```

Screenshot 2024-01-18 at 10 54 18 AM

#### Expected vs Current Behavior

LAD Tables are working as expected. LAD Table Sets should also load telemetry properly.

#### Steps to Reproduce
1. Create two SWGs
2. Create two LAD tables with above SWGs
3. Create a LAD Table Set with above LAD tables
4. Notice telemetry is always blank and exception is printed to console.

#### Impact Check List

- [x] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by scottbell**
Closes #7383 #7382

### Describe your changes:

- Ensure we provide `renderWhenVisible` to LadTableSets (and thus to child LadTables)
- Ensure TelemetryFrame always provides a telemetry keyString when opening a new tab
- Ensure `ladTelemetryObjects` is actually initialized before adding objects to it
- Wrote first (!) LadTableSet e2e test to ensure they work in the future

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Is this a breaking change to be called out in the release notes?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

### CODE REVIEW

**Comment by scottbell** [comment]
Tests are currently failing due to imagery missing. #7367 should resolve the issue.

File: e2e/tests/functional/plugins/lad/ladSet.e2e.spec.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      type: 'LAD Table',
+      parent: ladTableSet.uuid
+    });
+
+    await createDomainObjectWithDefaults(page, {
+      type: 'Sine Wave Generator',
+      parent: secondLadTable.uuid
+    });
+
+    await page.goto(ladTableSet.url);
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/lad/ladSet.e2e.spec.js:54]: move this into the testcase

File: e2e/tests/functional/plugins/lad/ladSet.e2e.spec.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  });
+
+  test('Ensure we have numbers in cells', async ({ page }) => {
+    const valueFromFirstSineWave = await page.locator('.js-third-data').first().innerText();
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/lad/ladSet.e2e.spec.js:58]: we're removing all page.locators in favor of a page.getBy to use user-facing attributes

File: e2e/tests/functional/plugins/lad/ladSet.e2e.spec.js (hunk 1 above)
Comment (ozyx) [reply e2e/tests/functional/plugins/lad/ladSet.e2e.spec.js:54]: @unlikelyzero @scottbell Note that `waitUntil: 'domcontentloaded'` is **only needed on initial page navigation**. If we're navigating via the router, we can use `page.goto(url)`, which is equivalent to `page.goto(url, { waitUntil: 'load' });`

**Review by ozyx — APPROVED** [review]
LGTM!

File: e2e/tests/functional/plugins/inspectorDataVisualization/numericData.e2e.spec.js [hunk 3]
```diff
@@ -36,7 +36,7 @@ test.describe('Testing numeric data with inspector data visualization (i.e., dat
     await page.goto('./', { waitUntil: 'domcontentloaded' });
   });
 
-  test('Can click on telemetry and see data in inspector', async ({ page }) => {
+  test('Can click on telemetry and see data in inspector', async ({ page, context }) => {
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/inspectorDataVisualization/numericData.e2e.spec.js:39]: this should have had a @2p

**Pull request merged**